pkg/api/validation/schema.go

```diff
@@ -219,8 +219,8 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, fieldName, typeName stri
 	return allErrs
 }
 
-// This matches type name in the swagger spec, such as "v1.Binding".
-var versionRegexp = regexp.MustCompile(`^v.+\..*`)
+// This matches type name in the swagger spec, such as "v1.Binding", "unversioned.ServerAddressByClientCIDR.
+var versionRegexp = regexp.MustCompile(`^(v.+|unversioned)\..*`)
```

Comment:
I talked to @lavalamp and he suggested that its better to copy ServerAddressByClientCIDR in federation types.go rather than using an unversioned struct there.
@jianhuiz Can you please send a PR for that.
We wont require this change then.